src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPSegment(POLY_OPA_DISP++, 0x08,
+                       Gfx_TexScroll(globalCtx->state.gfxCtx, 0, 0 - (globalCtx->gameplayFrames & 0x7F), 32, 32));
+            gSPDisplayList(POLY_OPA_DISP++, ovl_Boss_Ganon2_DL_0103A8);
+            if ((globalCtx->envCtx.unk_BD == 1) || (globalCtx->envCtx.unk_BD == 2)) {
+                alpha = (s16)(globalCtx->envCtx.unk_D8 * 150.0f) + 50;
+                angle = M_PI / 5.0f;
+            } else {
+                alpha = 100;
+                angle = M_PI / 2.0f;
+            }
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 0, 0, 0, alpha);
+            temp_f0 = effect->position.y - 1098.0f;
+            Matrix_Translate(effect->position.x + temp_f0, 1086.0f, (effect->position.z - 1.0f) + temp_f0, MTXMODE_NEW);
+            Matrix_RotateY(angle, MTXMODE_APPLY);
+            Matrix_Scale(1.0f, 0.0f, 1.0f, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 6155),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, ovl_Boss_Ganon2_DL_00F188);
+        }
+    }
+
+    effect = effects;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80903F38.s")
+    for (i = 0; i < 100; i++, effect++) {
+        if (effect->type == 2) {
+            if (spCD == 0) {
+                BossGanon2_SetObjectSegment(NULL, globalCtx, OBJECT_GEFF, true);
+                spCD++;
+            }
+            Matrix_Translate(effect->position.x, effect->position.y, effect->position.z, MTXMODE_NEW);
+            Matrix_Scale(effect->scale, effect->scale, effect->scale, MTXMODE_APPLY);
+            Matrix_RotateY(effect->unk_38.z, MTXMODE_APPLY);
+            Matrix_RotateX(effect->unk_38.y, MTXMODE_APPLY);
+            Matrix_RotateZ(effect->unk_38.x, MTXMODE_APPLY);
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 6179),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, D_06000EA0);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80904108.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 6185);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80904340.s")
+void func_80906538(BossGanon2* this, u8* shadowTexture, f32 arg2) {
+    s16 temp_t0;
+    s16 temp_v0;
+    s16 temp_a3;
+    s16 phi_v1;
+    s16 phi_a1;
+    s16 i;
+    f32 lerpx;
+    s16 j;
+    f32 lerpy;
+    f32 lerpz;
+    Vec3f sp70;
+    Vec3f sp64;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_8090464C.s")
+    for (i = 0; i < 15; i++) {
+        //! @bug j is not initialized if arg2 == 0.0f, causing undefined behavior.
```

Comment:
There's actually no bug here I think, if arg == 0 then the arg2 > 0 branch is never taken and all other branches either don't use j or initialize it

Reply:
You're right! I copied this comment from Sda https://github.com/zeldaret/oot/blob/master/src/overlays/actors/ovl_En_Sda/z_en_sda.c#L178 since the code was extremely similar but now that you mention it I don't think it has an issue either. Should I remove the comment from there as well?

Reply:
I guess, if it went through a PR once though I'd wait for a third opinion